syntax = "proto3";

package microless.socialnetwork.usertimeline;
option go_package = "microless/socialnetwork/proto/usertimeline";

import "proto/data.proto";
import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";
import "google/api/annotations.proto";

message WriteUserTimelineRequest {
    string post_id = 1;
    string user_id = 2;
    google.protobuf.Timestamp timestamp = 3;
}

message ReadUserTimelineRequest {
    string user_id = 1;
    int32 start = 2;
    int32 stop = 3;
}

message ReadUserTimelineRespond {
    repeated Post posts = 1;
}

message InsertUserResquest {
    string user_id = 1;
}

service UserTimelineService {
    rpc WriteUserTimeline (WriteUserTimelineRequest) returns(google.protobuf.Empty);
    rpc ReadUserTimeline (ReadUserTimelineRequest) returns(ReadUserTimelineRespond) {
        option (google.api.http) = {
            get: "/api/v1/usertimeline/{user_id}"
        };
    }
    rpc InsertUser (InsertUserResquest) returns(google.protobuf.Empty);
}